frontend/src/components/advanced-auto-filter/advanced-auto-filter.tsx

```diff
@@ -261,13 +277,45 @@ const AdvancedAutoFilter: FC<AdvancedAutoFilterProps> = (props) => {
             }))}
           />
         )}
-        <CheckboxList
-          title="Color"
-          list={options && options.colors}
-          checkedList={checkLists.colorId}
-          onListCheck={handleCheckboxListChange}
-          filterName={CheckListsNames.COLOR_ID}
-        />
+
+        <h5 className={styles.blockTitle}>Color</h5>
+        {options?.colors && (
+          <MemoizedMultiselectInput
+            label="Color"
+            filterName={CheckListsNames.COLOR_ID}
+            options={
+              options &&
+              options?.colors.map((item: AutoRiaOption) => ({
+                label: item.name,
+                id: item.id,
+              }))
+            }
+            value={checkLists.colorId.map((id) =>
+              getValueById(options.colors, id),
+            )}
+            onChange={handleCheckboxListChange}
+          />
+        )}
+
+        <h5 className={styles.blockTitle}>Fuel</h5>
+        {options?.fuelTypes && (
+          <MemoizedMultiselectInput
+            label="Fuel"
+            filterName={CheckListsNames.FUEL_TYPE_ID}
+            options={
+              options &&
+              options?.fuelTypes.map((item: AutoRiaOption) => ({
+                label: item.name,
+                id: item.id,
+              }))
+            }
+            value={checkLists.fuelTypeId.map((id) =>
+              getValueById(options.fuelTypes, id),
+            )}
```

Comment:
Usually, we mustn't pass the `.map` result as the prop. Because `.map` **always returns a new reference**. So it breaks the `memo()`